components/clp-credential-manager/src/config.rs

```diff
@@ -0,0 +1,100 @@
+use std::path::Path;
+
+use clp_rust_utils::{Error as UtilsError, serde::yaml};
+use secrecy::SecretString;
+use serde::Deserialize;
+
+use crate::error::{ServiceError, ServiceResult};
+
+const DEFAULT_MAX_CONNECTIONS: u32 = 5;
+const DEFAULT_SERVER_BIND_ADDRESS: &str = "0.0.0.0";
+const DEFAULT_SERVER_PORT: u16 = 8080;
+
+/// Root configuration deserialized from `credential-manager-config.yml`.
+#[derive(Debug, Clone, Deserialize)]
+pub struct AppConfig {
+    #[serde(default)]
+    pub server: ServerConfig,
+    pub database: DatabaseConfig,
+}
+
+impl AppConfig {
+    /// Loads configuration from disk and validates YAML syntax.
+    ///
+    /// # Parameters:
+    ///
+    /// * `path`: Filesystem location of the YAML configuration file.
+    ///
+    /// # Returns:
+    ///
+    /// A parsed [`AppConfig`] instance when the file can be read and deserialized successfully.
+    ///
+    /// # Errors:
+    ///
+    /// * Returns [`ServiceError::Io`] if the file cannot be read.
+    /// * Returns [`ServiceError::Yaml`] if parsing fails.
+    pub fn from_file(path: &Path) -> ServiceResult<Self> {
+        let config: Self = yaml::from_path(path)?;
+        Ok(config)
+    }
+}
+
+/// Network settings for the Axum server.
+#[derive(Debug, Clone, Deserialize)]
+#[serde(default)]
+pub struct ServerConfig {
+    pub bind_address: String,
+    pub port: u16,
+}
+
+impl Default for ServerConfig {
+    /// Provides sensible defaults that bind the HTTP server to all interfaces on port 8080.
+    fn default() -> Self {
+        Self {
+            bind_address: default_bind_address(),
+            port: default_bind_port(),
+        }
+    }
+}
+
+/// Supplies the default bind address of `0.0.0.0` so the service listens on every interface.
```

Comment:
```suggestion
/// Returns the default bind address of `0.0.0.0` so the service listens on every interface.
```
Prefer `Returns`. Do it for other ones.